userland/examples/tutorials/xx_ble_adv/main.c

```diff
@@ -0,0 +1,93 @@
+#include <ble.h>
+#include <led.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <temperature.h>
+#include <timer.h>
+#include <tock.h>
+#include <unistd.h>
+
+const char* get_error(int err);
+
+/*******************************************************************************
+ * MAIN
+ ******************************************************************************/
+
+int main (void) {
+  int err;
+  printf("[Tutorial] BLE Advertising\n");
+
+  // BLE DATA FIELD
+  uint8_t addr[] = {1, 2, 3, 4, 5, 6};
+  uint16_t advertising_itv_ms = 300;
+  uint8_t device_name[]       = "TockOS";
+  uint16_t uuids[] = {0x1800, 0x1809};
+  uint8_t manufacturer_data[]     = {0x13, 0x37};
+  uint8_t fake_temperature_data[] = {0x00, 0x00};
+
+  // configure advertisement interval to 300ms
+  // configure LE only
+  // configure advertisement address as 1,2,3,4,5,6
+  err = ble_initialize(advertising_itv_ms,
+                       LE_LIMITED_DISCOVERABLE | BREDR_NOT_SUPPORTED,
+                       addr,
+                       sizeof(addr));
```

Comment:
Technically, `sizeof` on buffers allocated on the local stack does work successfully. However, I'd argue that it's a bad practice to do so. In the general case, `sizeof` does not work on buffers, and using them this way leads to mistakes (like someone making a buffer global instead of local and code suddenly breaking).

I'd suggest replacing them with explicit lengths.

Reply:
Hmm, is this explanation really correct? I thought it has nothing to do whether there are stored on stack or in data segment since there are evaluated at compile-time? However, since arrays are automatically converted to pointers upon a function call you can not us sizeof then because you get the word size then. But send a link if I'm wrong this got me curious.

Reply:
Doing some research shows you to be correct Niklas. `sizeof` works on any named buffer. So like you say, it fails when used on a buffer is passed through a function call, but works on any local buffer on the stack or global buffer in the same file.

I've always learned it to be a bad practice though, probably because beginning programmers do not notice the difference, and using `sizeof` on the other side of a function boundary is a common mistake. I could see the argument that it is appropriate here.